app/controllers/api/v1/application_users_controller.rb

```diff
@@ -92,6 +92,33 @@ def index
   end
 
   ###############################################################
+  # username
+  ###############################################################
+
+  api :GET, '/application_users/username/:username',
+      'Gets a single User with the specified username.'
+  description <<-EOS
+      All users of OpenStax have an associated User object.  This endpoint
+      allows querying additional data on a user when all that is known is the
+      User's username.
+
+      Admins (for Accounts only) are identified by the is_administrator boolean.
+      Some additional user information can be found in associations, such as
+      email addresses in ContactInfos and the password hash in Identity.
+
+      Users have the following String attributes:
+      username, first_name, last_name, full_name, title
+
+    #{json_schema(Api::V1::UserSearchRepresenter, include: :readable)}
+  EOS
+  def username
+    OSU::AccessPolicy.require_action_allowed!(:read, current_api_user, ApplicationUser)
```

Comment:
To enforce that the current_api_user (hopefully an application) can actually read the ApplicationUser being retrieved, that ApplicationUser should be retrieved first and then passed in to the `require_action_allowed!`, e.g.

```
def username
  application_user = ApplicationUser.includes(:user).joins(:user).where{user.username.eq username}
  OSU::AccessPolicy.require_action_allowed!(:read, current_api_user, application_user)
  respond_with application_user, represent_with: Api::V1::ApplicationUserRepresenter
end
```

You might even be able to use `standard_read` (then again, Account's version of openstax_api may be too old).